### PR TITLE
Reindex lambda and move alias

### PR DESCRIPTION
**Pull request by elcorbs**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-411

### *What is the problem we're trying to solve*
Two problems 
1. We are using DMS to migrate the data, meaning we have no control over the resulting index settings. We need to be able to decide the mappings to utilise custom analysers. 
2. We need a process to change the index settings if we need to. 

### *What changes have we introduced*

Created two lambda functions which work as follow
- Lambda 1 
 You can pass this lambda an alias name, an index to reindex from (if not provided it will find the index currently attached to the alias), and index config in json format (if not provided it will use the index.json file stored in the address API data folder)
1, It will create a new index using the config and the name will be the alias + current timestamp
2, It will then start reindexing the data on the elasticsearch server.
3, It will send the TaskId for the reindex to the sqs queue with a 10 minute delay 
(On testing, in took around 7m to reindex hackney addresses and around 20 mins to reindex national addresses)

- Lambda 2
Trigger by the sqs queue event
1, It will get the task for the task Id.
2, If the task has not completed, it will the message back to the queue with a 10 minute delay
3, if the tasks has completed, it will remove the alias name from any current index that have it and add the alias name to the newly creating index. If there are error from reindexing it will write them to the log.
4, Deletes the task document.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

- Add documentation around this process.
- Add an option to delete the initial index after reindex.
- dynamically populate account number in serverless.yml

### CODE REVIEW

File: Reindex/Handler.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+            await RemoveAllReferencesToAlias(data.alias);
+
+            await _elasticSearchClient.Indices.PutAliasAsync(Indices.Index(data.newIndex), data.alias);
+            LambdaLogger.Log($"Address alias {data.alias} to index {data.newIndex}");
+
+            await _elasticSearchClient.DeleteAsync<TaskId>(data.taskId, d => d.Index(".tasks"));
+            LambdaLogger.Log($"Deleted Task document with ID {data.taskId}");
+
+            //Delete old index?
```
Comment (Renny4Real) [inline Reindex/Handler.cs:122]: Is this a `TODO`, maybe after testing in Staging?

Comment (elcorbs) [reply Reindex/Handler.cs:122]: Yep, I might make a ticket for it

File: Reindex/Handler.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                await _elasticSearchClient.Indices.DeleteAliasAsync(Indices.Index(index), alias);
+                LambdaLogger.Log($"Removed alias {alias} from index {index}");
+            }
+        }
+
+        private async Task<SendMessageResponse> SendSqsMessageToQueue(string message)
+        {
+            var sqsRequest = new SendMessageRequest
+            {
+                DelaySeconds = 600,
```
Comment (crablab) [inline Reindex/Handler.cs:139]: Should this be configurable? Perhaps via a default parameter into the method?

**Review by crablab — APPROVED** [review]

File: Reindex/Handler.cs (hunk 2 above)
Comment (elcorbs) [reply Reindex/Handler.cs:139]: Yep, good shout.

**Pull request merged**